src/Network/Network/Cortex/VHubRouteTable/UpdateAzureRmVHubRouteTableCommand.cs

```diff
@@ -0,0 +1,161 @@
+            HelpMessage = "The resource id route table to modify.")]
+        [ResourceIdCompleter("Microsoft.Network/virtualHubs/hubRouteTables")]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The list of routes for this route table resource.")]
+        public PSVHubRoute[] Route { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "List of labels for this route table resource.")]
+        public string[] Label { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+            PSVHubRouteTable hubRouteTableToUpdate = null;
+            if (ParameterSetName.Equals(CortexParameterSetNames.ByVHubRouteTableObject, StringComparison.OrdinalIgnoreCase))
+            {
+                hubRouteTableToUpdate = this.InputObject;
+                this.ResourceId = this.InputObject.Id;
+                if (string.IsNullOrWhiteSpace(this.ResourceId))
+                {
+                    throw new PSArgumentException(Properties.Resources.VHubRouteTableNotFound);
+                }
+
+                var parsedResourceId = new ResourceIdentifier(this.ResourceId);
+                this.ResourceGroupName = parsedResourceId.ResourceGroupName;
+                this.ParentResourceName = parsedResourceId.ParentResource.Split(new[] { '/' }, StringSplitOptions.RemoveEmptyEntries).Last();
+                this.Name = parsedResourceId.ResourceName;
+            }
+            else
+            {
+                if (ParameterSetName.Equals(CortexParameterSetNames.ByVHubRouteTableResourceId, StringComparison.OrdinalIgnoreCase))
+                {
+                    var parsedResourceId = new ResourceIdentifier(this.ResourceId);
+                    this.ResourceGroupName = parsedResourceId.ResourceGroupName;
+                    this.ParentResourceName = parsedResourceId.ParentResource.Split(new[] { '/' }, StringSplitOptions.RemoveEmptyEntries).Last();
+                    this.Name = parsedResourceId.ResourceName;
+                }
+                else if (ParameterSetName.Equals(CortexParameterSetNames.ByVirtualHubObject, StringComparison.OrdinalIgnoreCase))
+                {
+                    var parentResourceId = this.ParentObject.Id;
+                    var parsedParentResourceId = new ResourceIdentifier(parentResourceId);
+                    this.ResourceGroupName = parsedParentResourceId.ResourceGroupName;
+                    this.ParentResourceName = parsedParentResourceId.ResourceName;
+                }
+
+                hubRouteTableToUpdate = this.GetVHubRouteTable(this.ResourceGroupName, this.ParentResourceName, this.Name);
+                if (hubRouteTableToUpdate == null)
```

Comment:
if this vhub route table was not found, will it terminate and throw exceptions?

Reply:
yes, is it not expected?

Reply:
I think line 131 will throw exception before line 132 was executed.

You might want to add a try block for line 131 and catch the exception

Reply:
handling it with try-catch in GetVHubRouteTable method itself.